internal/provider/provider.go

```diff
@@ -451,12 +451,12 @@ func NewFrameworkProvider() provider.Provider {
 }
 
 func MuxedProviderFactory() func() tfprotov6.ProviderServer {
-	return MuxedProviderFactoryFn(NewSdkV2Provider())
+	return muxedProviderFactoryFn(NewSdkV2Provider())
 }
 
-// MuxedProviderFactoryFn creates mux provider using existing sdk v2 provider passed as parameter and creating new instance of framework provider.
+// muxedProviderFactoryFn creates mux provider using existing sdk v2 provider passed as parameter and creating new instance of framework provider.
 // Used in testing where existing sdk v2 provider has to be used.
-func MuxedProviderFactoryFn(sdkV2Provider *sdkv2schema.Provider) func() tfprotov6.ProviderServer {
+func muxedProviderFactoryFn(sdkV2Provider *sdkv2schema.Provider) func() tfprotov6.ProviderServer {
```

Comment:
Since this function receiving an existing sdkV2Provider is no longer needed, could we define this logic directly in `MuxedProviderFactory()` and remove this function?